overlord/snapstate/snapstate.go

```diff
@@ -287,6 +287,19 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	// check if we already have the revision locally (alters tasks)
 	revisionIsLocal := snapst.LastIndex(targetRevision) >= 0
 
+	// We will not need the temporary snap if we have a local revision.
+	// This can happen when e.g. side-loading a local revision again.
+	// The SnapPath is only needed in the "mount-snap" handler and
+	// that is skipped for local revisions.
+	//
+	// XXX: should this be closer to where we skip generating the mount
+	//      task?
```

Comment:
naively I would say so but I haven't looked a the code

Reply:
I looked at the code, there is not much code in between, so it doesn't matter much.  I would think putting it closer to the mount task is still a win if any of this needs later changes and the extra time window looks tiny.

Reply:
Thanks, I moved it now